org.jcryptool.visual.rabin/src/org/jcryptool/visual/rabin/HandleFirstTab.java

```diff
@@ -0,0 +1,2075 @@
+package org.jcryptool.visual.rabin;
+
+import java.math.BigInteger;
+import java.security.SecureRandom;
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.Random;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.custom.ScrolledComposite;
+import org.eclipse.swt.events.MouseEvent;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.events.VerifyListener;
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Combo;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Text;
+import org.eclipse.ui.IEditorInput;
+import org.eclipse.ui.PartInitException;
+import org.jcryptool.core.operations.editors.AbstractEditorService;
+import org.jcryptool.core.operations.editors.EditorsManager;
+import org.jcryptool.core.util.colors.ColorService;
+import org.jcryptool.crypto.ui.textloader.ui.wizard.TextLoadController;
+import org.jcryptool.visual.rabin.ui.RabinFirstTabComposite;
+import org.jcryptool.visual.rabin.ui.RabinSecondTabComposite;
+
+
+public class HandleFirstTab extends GUIHandler {
+	
+	private String strOnlyDecAllowed = Messages.HandleFirstTab_strOnlyDecAllowed;
+	private String strUpperLimitRestriction = Messages.HandleFirstTab_strUpperLimitRestriction;
+	private String strLowerLimitLessUpperLimit = Messages.HandleFirstTab_strLowerLimitLessUpperLimit;
+	private String strAppropriatePrimesWarning = Messages.HandleFirstTab_strAppropriatePrimesWarning;
+	public boolean stopComputation = false;
+	public String oldN = null;
+	
+
+//	/**
+//	 * @param scMain
+//	 * @param compMain
+//	 * @param rabinFirst
+//	 * @param rabinSecond
+//	 */
+//	public HandleFirstTab(ScrolledComposite scMain, Composite compMain, Rabin rabinFirst, Rabin rabinSecond) {
+//		super(scMain, compMain, rabinFirst, rabinSecond);
+//	}
+	
+	/**
+	 * @param scMain
+	 * @param compMain
+	 * @param rabinFirst
+	 * @param rabinSecond
+	 */
+	public HandleFirstTab(ScrolledComposite scMain, Composite compMain, Rabin rabin) {
+		super(scMain, compMain, rabin);
+	}
+	
+	
+	
+	
+	
+	
+	/**
+	 * parses strings of the form decmial^decimal
+	 * @param limit
+	 * @return parsed integer of the form decimal^decimal
+	 */
+	private BigInteger getParsedNumFromLimit(String limit) {
+		BigInteger ret = null;
+		String pattern = "\\^"; //$NON-NLS-1$
+		
+		String[] parsedNum = limit.split(pattern);
+		BigInteger base = new BigInteger(parsedNum[0]);
+		int exp = Integer.parseInt(parsedNum[1]);
+		ret = base.pow(exp);
+		
+		return ret;
+		
+	}
+	
+	
+	
+	/**
+	 * @param limit
+	 * @return converted string limit to BigInteger
+	 */
+	private BigInteger getNumFromLimit(String limit) {
```

Comment:
It seems this method does actually two things. It seems this is not needed, as the callers know which functionality they need.